### PR TITLE
Redid solver to be a simple program

### PR DESCRIPTION
**Pull request by Maje419**
Running the image will still fail outside of a Kubernetes cluster, but the code works just trust me

### CODE REVIEW

File: Solver/Solver.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    has_dzn_file: True || False             - Describes whether a DZN file will be provided
+    solver_name: 'chuffed' || 'geode'       - The name of the solver to use
+    mzn_file: string                        - The url of the mzn_file
+    dzn_file: string                        - If hasDznFile was true, the url of the dzn_file
+    number_processors : int                 - The number of processors to use
+'''
+
+class SolverInstance:
+
+    userID = "testUser"
```
Comment (TotallyRealHolm) [inline Solver/Solver.py:22]: That is a lot of comments

Comment (darkIvor) [reply Solver/Solver.py:22]: where is a lot of comments

Comment (TotallyRealHolm) [reply Solver/Solver.py:22]: right here

Comment (darkIvor) [reply Solver/Solver.py:22]: but there are no arguments either.

File: Solver/Solver.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            timer += 1
+            bestResult = i
+
+        self.notify_final_solution_found(bestResult)
+
+    def __init__(self, args):
+        if (len(args) > 1):
+            self.userID = args[1]
+            self.taskID = args[2]
+            self.has_dzn_file = args[3].upper() == "TRUE"
```
Comment (TotallyRealHolm) [inline Solver/Solver.py:84]: Modern problems require modern solutions

Comment (TotallyRealHolm) [reply Solver/Solver.py:84]: Try typecasting it with bool()

File: ServiceTests/SolverTests/SolverTests.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+namespace ServiceTests;
+
+[TestClass]
+public class SolverTests
+{
+    [TestMethod]
+    public void HealthCheck_Works()
+    {
```
Comment (TotallyRealHolm) [inline ServiceTests/SolverTests/SolverTests.cs]: Remember to remove the C sharp project files

**Review by TotallyRealHolm — CHANGES_REQUESTED** [review]
Remember to remove the charp project files

File: Solver/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,25 @@
+FROM python:3.8-slim-buster
+
+WORKDIR "/app"
+
+COPY Solver.py .
+COPY requirements.txt .
+COPY Examples/shortest.mzn Test/shortest.mzn
```
Comment (TotallyRealHolm) [inline Solver/Dockerfile:7]: We need to remember to remove these , for the final image

File: Solver/Solver.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    def notify_intermediate_solution_found(self, result: Result):
+        result_as_json = self.get_result_as_json(result)
+        requests.post(self.solution_manager_url + "/SolutionFound", data=result_as_json)
+
+    def notify_final_solution_found(self, result: Result):
+        result_as_json = self.get_result_as_json(result, True)
+        requests.post(self.solver_manager_url + "/Solution/{taskID}".format(taskID = self.taskID), data=json.dumps({"UserID" : self.userID}))
+        requests.post(self.solution_manager_url + "/SolutionFound", data=result_as_json)
+    
+    #This function is here to download the files at some point
```
Comment (TotallyRealHolm) [inline Solver/Solver.py:46]: Might be good with TODO tag

**Review by Maje419 — COMMENTED** [review]
The ServiceTests have been removed

**Review by TotallyRealHolm — CHANGES_REQUESTED** [review]
Suggested some changes, after that LGTM.

File: Solver/Solver.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        self.userID = os.getenv("USERID") if os.getenv("USERID") else "testUser"
+        
+        self.taskID = os.getenv("TASKID") if os.getenv("TASKID") else "testTask"
+        self.solver_manager_service = os.getenv("SOLVER_MANAGER_SERVICE") if os.getenv("SOLVER_MANAGER_SERVICE") else ("0.0.0.0")
+        self.solution_manager_service = os.getenv("SOLUTION_MANAGER_SERVICE") if os.getenv("SOLUTION_MANAGER_SERVICE") else ("0.0.0.0")
+        
+        self.mzn_file = args[1]
+        self.solver_name = args[2]
+        self.number_processors = args[3]
+        if (len(args) == 5):
```
[the remark this answers is not in the document]
Comment (Maje419) [reply Solver/Solver.py:90]: Good call. The DZN file needs to be the last argument since it is optional, so i moved the MZN file.

**Review by TotallyRealHolm — APPROVED** [review]
LGTM

**Pull request merged**